packages/framework-types/src/searcher.ts

```diff
@@ -114,11 +129,40 @@ export class Searcher<TObject> {
       this._sortByList,
       this._limit,
       this._afterCursor,
-      this._paginatedVersion
+      this._paginatedVersion,
+      this._selectFor,
+      this._skipInstance
     )
   }
 }
 
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+// type KeyOf<TType> = {
+//   [K in keyof TType & (string | number)]: TType[K] extends object ? `${K}` | `${K}.${KeyOf<TType[K]>}` : `${K}`
+// }[keyof TType & (string | number)]
+
+// Explicitly depth-limit recursive type so that by default things only descend four levels
+type Prev = [never, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+type KeyOf<TType, D extends Prev[number] = 4> = [D] extends [never]
+  ? never
+  : {
+      [K in keyof TType & (string | number)]: TType[K] extends object
+        ? `${K}` | `${K}.${KeyOf<TType[K], Prev[D]>}`
+        : `${K}`
+    }[keyof TType & (string | number)]
+
+// type KeyOf<
+//   TType,
+//   R = {
+//     [K in keyof TType & (string | number | bigint | boolean | null | undefined)]: TType[K] extends object
+//       ? `${K}` | `${K}.${KeyOf<TType[K]>}`
+//       : `${K}`
+//   }[keyof TType & (string | number)]
+// > = R
+
+export type ProjectionFor<TType> = Array<KeyOf<TType>>
```

Comment:
If I understood correctly, this types is an array of all the properties of the object, including sub-properties. 

If so, this type can consume a lot of memory and time when compiling. Newer versions of Typescript allows you to do some easier black magic which is much much faster and requires much less memory (up to 6 times less)

```suggestion
type Paths<T> = T extends object ? { [K in keyof T]: `${Exclude<K, symbol>}${'' | `.${Paths<T[K]>}`}` }[keyof T] : never

export type ProjectionFor<TType> = Array<Paths<TType>>
```

Reply:
Suggestion applied in commit e2d6ac5.